### PR TITLE
fix CCConsole linux crash

### PR DESCRIPTION
**Pull request by heliclei**

### CODE REVIEW

**Comment by heliclei** [comment]
[ci rebuild]

File: tests/Classes/ConsoleTest/ConsoleTest.cpp [hunk 1]
```diff
@@ -187,6 +187,7 @@ void ConsoleCustomCommand::onEnter()
 
    _src_file_path = FileUtils::getInstance()->fullPathForFilename(s_pathGrossini);
     _thread = std::thread( &ConsoleUploadFile::uploadFile, this);
+    _thread.detach();
```
Comment (dumganhar) [inline tests/Classes/ConsoleTest/ConsoleTest.cpp]: If you use `detach`, member variable `_thread` will not be needed.

Comment (heliclei) [inline tests/Classes/ConsoleTest/ConsoleTest.cpp]: thanks, I will remove it

File: cocos/base/CCConsole.cpp [hunk 2]
```diff
@@ -971,6 +972,17 @@ void Console::loop()
             for(const auto &fd: _fds) {
                 if(FD_ISSET(fd,&copy_set)) 
                 {
+//fix Bug #4302 Test case ConsoleTest--ConsoleUploadFile crashed on Linux
+#if defined(LINUX)
```
Comment (dumganhar) [inline cocos/base/CCConsole.cpp]: Please use `#if (CC_TARGET_PLATFORM == CC_PLATFORM_LINUX)`

File: cocos/base/CCConsole.cpp [hunk 3]
```diff
@@ -971,6 +972,17 @@ void Console::loop()
             for(const auto &fd: _fds) {
                 if(FD_ISSET(fd,&copy_set)) 
                 {
+//fix Bug #4302 Test case ConsoleTest--ConsoleUploadFile crashed on Linux
```
Comment (minggo) [inline cocos/base/CCConsole.cpp:974]: Could you please write the reason that crashed on linux here?
Thanks.

**Comment by minggo** [comment]
And please fix conflict.

File: cocos/base/CCConsole.cpp (hunk 3 above)
Comment (heliclei) [inline cocos/base/CCConsole.cpp:974]: Thanks, I will add the detailed notes

**Comment by cocosQA** [comment]
Dear heliclei:
  We have created a new branch 'v3' to replace branch 'develop', and this PR has been moved from 'develop' branch to 'v3' branch, the new PR is https://github.com/cocos2d/cocos2d-x/pull/6453
The old 'develop' branch will be deleted soon, and this PR will be closed accordingly.
